homeassistant/components/automation/config.py

```diff
@@ -51,7 +51,7 @@ async def _try_async_validate_config_item(hass, config, full_config=None):
         IntegrationNotFound,
         InvalidDeviceAutomationConfig,
     ) as ex:
-        async_log_exception(ex, DOMAIN, full_config or config, hass)
+        async_log_exception(ex, DOMAIN, config or full_config, hass)
```

Comment:
`async_log_exception` expects the full config to be passed in since it will try to extract the domain config. I think that we should update `async_log_exception` to not be smart and expect the bad config to be passed in.

Reply:
To me my change seems to work just fine. 

**Original code** (full_config being used) => error can only point to the line in `configuration.yaml` where I reference the `automations.yaml`.
> 2020-10-19 21:27:09 ERROR (MainThread) [homeassistant.config] Invalid config for [automation]: [trigger123] is an invalid option for [automation]. Check: automation->trigger123. (See /home/philip/.homeassistant/configuration.yaml, line 5).

**My change** (config being preferred) => the correct config is provided and hence the specific error can be logged:

> 2020-10-19 21:28:27 ERROR (MainThread) [homeassistant.config] Invalid config for [automation]: [trigger123] is an invalid option for [automation]. Check: automation->trigger123. (See /home/philip/.homeassistant/automations.yaml, line 19).

Reply:
Right… but the code is not used as it is intended. `async_log_exception` will do `config[DOMAIN]` and that's wrong in this case.